src/main/java/com/algorand/algosdk/abi/Interface.java

```diff
@@ -0,0 +1,44 @@
+package com.algorand.algosdk.abi;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class Interface {
+    @JsonProperty("name")
+    public String name;
+    @JsonProperty("methods")
+    public List<Method> methods = new ArrayList<>();
+
+    @JsonCreator
+    public Interface(
+            @JsonProperty("name") String name,
+            @JsonProperty("methods") List<Method> methods
+    ) {
+        this.name = Objects.requireNonNull(name, "name must not be null");
+        if (methods != null) this.methods = methods;
+    }
+
+    @JsonIgnore
```

Comment:
well, the weird thing in Jackson Json annotation is that, if you have `get`-alike function name, it will store unnecessary stuff in encoded Json string.

Reply:
Ok, that makes sense, thanks!